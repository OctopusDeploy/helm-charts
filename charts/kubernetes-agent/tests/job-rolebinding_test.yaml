suite: "job permissions"
templates:
- templates/job-rolebinding.yaml
tests:
- it: "creates a role binding per target namespace"
  set:
    jobServiceAccount:
      targetNamespaces: ["ns-1", "ns-2", "ns-3"]
  asserts:
  - hasDocuments:
      count: 3

- it: "matches snapshot with one target namespace"
  set:
    jobServiceAccount:
      targetNamespaces: ["ns-1"]
  asserts:
  - matchSnapshot: {}

- it: "is not created when target namespaces is empty"
  set:
    jobServiceAccount:
      targetNamespaces: []
  asserts:
  - hasDocuments:
      count: 0

- it: "is not created by default"
  asserts:
  - hasDocuments:
      count: 0
