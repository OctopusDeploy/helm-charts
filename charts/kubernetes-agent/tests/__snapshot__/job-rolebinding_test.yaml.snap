matches snapshot with one target namespace:
  1: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: RoleBinding
    metadata:
      name: octopus-agent-job-RELEASE-NAME-binding
      namespace: ns-1
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: ClusterRole
      name: octopus-agent-job-RELEASE-NAME-role
    subjects:
      - kind: ServiceAccount
        name: octopus-agent-job
        namespace: NAMESPACE
